### ISSUE TITLE
The UI thread hangs around

### ISSUE THREAD

**Issue by bjourne** [factor/factor#1905]
Something the UI thread hangs around even after the UI has been terminated. For example, try running `nehe.4:run4` and then close the gl window. `ui-running?` will be  `f` which is right, but when listing the threads using `threads.` the UI thread will be there. It's not a big problem because the thread is just waiting on a flag, but it is a little unclean I think.

**Comment by mrjbq7** [comment]
Was this fixed by f93da595f5091ae23f6a641b9ec945c10e848caf?

**Comment by bjourne** [comment]
No, that fix was for the ui thread duplicating. This bug is for the ui thread not dying when you close all windows.
